synapse/config/registration.py

```diff
@@ -104,8 +104,8 @@ def read_config(self, config, **kwargs):
 
         self.bcrypt_rounds = config.get("bcrypt_rounds", 12)
         self.trusted_third_party_id_servers = config.get(
-            "trusted_third_party_id_servers", ["matrix.org", "vector.im"]
-        )
+            "trusted_third_party_id_servers"
+        ) or ["matrix.org", "vector.im"]
```

Comment:
I'm not sure this is correct. It will mean that if `trusted_third_party_id_servers` is deliberately configured to the empty list, we will override that choice with `["matrix.org", "vector.im"]`.

Reply:
Ah, true. I changed it because we I assumed we didn't handled the case where `trusted_third_party_id_servers` could be None, but looking at the code each time it's invoked is surrounded in ` if trusted_third_party_id_server`.

I'll change it back.